### PR TITLE
Postgres Extensions dashboard

### PR DESCRIPTION
**Pull request by nasc17**
Show list of extensions
Enables adding/dropping extensions

image
image

### CODE REVIEW

File: extensions/arc/src/common/utils.ts [hunk 1]
```diff
@@ -185,6 +185,22 @@ export async function promptAndConfirmPassword(validate: (input: string) => stri
 	return undefined;
 }
 
+/**
+ * Opens an input box
+ * @param validate A function that accepts value and returns an error message if it's invalid
+ * @returns Promise resolving to the value if it passed validation,
+ * or undefined if the input box was closed for any other reason
+ */
+export async function promptValue(header: string, prompt: string, validate: (input: string) => string): Promise<string | undefined> {
```
Comment (Charles-Gagnon) [inline extensions/arc/src/common/utils.ts:194]: This doesn't seem like something we really need a separate function for given that you can just call promptInputBox with the specified options directly (it's not doing any logic that would make this useful)

File: extensions/arc/src/ui/dashboards/postgres/postgresExtensionsPage.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+					headerCssStyles: cssStyles.tableHeader,
+					rowCssStyles: cssStyles.tableRow
+				}
+			],
+			data: []
+		}).component();
+
+		this.extensionsLoading = this.modelView.modelBuilder.loadingComponent()
+			.withItem(this.extensionsTable)
+			.withProperties<azdata.LoadingComponentProperties>({
```
Comment (Charles-Gagnon) [inline extensions/arc/src/ui/dashboards/postgres/postgresExtensionsPage.ts:96]: Suggest adding a `loadingText` and `loadingCompletedText` that includes what is being loaded

File: extensions/arc/src/ui/dashboards/postgres/postgresExtensionsPage.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+								location: vscode.ProgressLocation.Notification,
+								title: loc.updatingInstance(this._postgresModel.info.name),
+								cancellable: false
+							},
+							async (_progress, _token): Promise<void> => {
+								try {
+									await this._azdataApi.azdata.arc.postgres.server.edit(
+										this._postgresModel.info.name,
+										{
+											extensions: extensions
```
Comment (Charles-Gagnon) [inline extensions/arc/src/ui/dashboards/postgres/postgresExtensionsPage.ts:130]: Is this going to just add the specified extensions or will it remove ones that aren't in this list?

File: extensions/arc/src/ui/dashboards/postgres/postgresExtensionsPage.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+										},
+										this._postgresModel.controllerModel.azdataAdditionalEnvVars);
+								} catch (err) {
+									// If an error occurs while editing the instance then re-enable the save button since
```
Comment (Charles-Gagnon) [inline extensions/arc/src/ui/dashboards/postgres/postgresExtensionsPage.ts:134]: You can just use finally for this if you're rethrowing the error

Although on line 152 you're already setting this to enabled after this is all done anyways - so is this even necessary?

**Comment by nasc17** [comment]
Separate viewing and editing extensions to separate PRs. 
View: https://github.com/microsoft/azuredatastudio/pull/15887

**Pull request closed without merging**